Specs/Scene/SceneSpec.js

```diff
@@ -838,6 +838,43 @@ defineSuite([
         });
     });
 
+    it('pickPosition caches results per frame',function(){
+        var s = createScene();
+        if (!s.pickPositionSupported) {
+            return;
+        }
+
+        var rectangle = Rectangle.fromDegrees(-100.0, 30.0, -90.0, 40.0);
+        s.camera.setView({ destination : rectangle });
+
+        var canvas = s.canvas;
+        var windowPosition = new Cartesian2(canvas.clientWidth / 2, canvas.clientHeight / 2);
+        spyOn(s, 'pickPositionWorldCoordinates').and.callThrough();
+
+        expect(s).toRenderAndCall(function() {
+            s.pickPosition(windowPosition);
+            expect(s.pickPositionWorldCoordinates).toHaveBeenCalled();
+
+            s.pickPosition(windowPosition);
+            expect(s.pickPositionWorldCoordinates.calls.count()).toEqual(1);
+
+            var rectanglePrimitive = createRectangle(rectangle);
+            rectanglePrimitive.appearance.material.uniforms.color = new Color(1.0, 0.0, 0.0, 1.0);
+
+            var primitives = s.primitives;
+            primitives.add(rectanglePrimitive);
+        });
+
+        expect(s).toRenderAndCall(function() {
+            s.pickPosition(windowPosition);
+            expect(s.pickPositionWorldCoordinates.calls.count()).toEqual(2);
+
+            s.pickPosition(windowPosition);
+            expect(s.pickPositionWorldCoordinates.calls.count()).toEqual(2);
+        });
+
```

Comment:
First pickPosition calculates and returns undefined. The second one returns undefined from cache. Third calculates a valid Cartesian3 then forth one returns valid value from cache. So both calls are needed to test correctly. Because the flow of undefined and valid results are seperate.